api/v1alpha1/progressiverollout_types.go

```diff
@@ -28,18 +30,68 @@ type ProgressiveRolloutSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	// Foo is an example field of ProgressiveRollout. Edit ProgressiveRollout_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// SourceRef references the resource, example an ApplicationSet, which owns ArgoCD Applications
+	//+kubebuilder:validation:Required
+	SourceRef corev1.TypedLocalObjectReference `json:"sourceRef"`
+	// Stages reference a list of Progressive Rollout stages
+	//+kubebuilder:validation:Optional
+	Stages []*ProgressiveRolloutStage `json:"stages,omitempty"`
+}
+
+// ProgressiveRolloutStage defines a rollout stage
+type ProgressiveRolloutStage struct {
+	// Name is a human friendly name for the stage
+	//+kubebuilder:validation:Required
+	Name string `json:"name"`
+	// MaxParallel is how many selected targets to update in parallel
+	//+kubebuilder:validation:Minimum:1
+	MaxParallel intstr.IntOrString `json:"maxParallel"`
```

Comment:
`IntOrString` is this an option? or is that to allow things like `100%`?

Reply:
That is coming from https://pkg.go.dev/k8s.io/apimachinery/pkg/util/intstr@v0.20.1. It's used as well in the deployment controller so you can specify an integer or a %, see https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/deployment/util/deployment_util.go

Reply:
as i suspected. thank you for clarifying.